x-pack/filebeat/input/httpjson/input.go

```diff
@@ -5,596 +5,152 @@
-			if msg.StatusCode == http.StatusTooManyRequests {
-				if err = in.applyRateLimit(ctx, header, in.config.RateLimit); err != nil {
-					return err
-				}
-				continue
-			}
-			return errors.Errorf("http request was unsuccessful with a status code %d", msg.StatusCode)
-		}
-
-		err = json.Unmarshal(responseData, &m)
-		if err != nil {
-			in.log.Debug("failed to unmarshal http.response.body", string(responseData))
-			return errors.Wrapf(err, "failed to unmarshal http.response.body")
-		}
-		switch obj := m.(type) {
-		// Top level Array
-		case []interface{}:
-			mm, err = in.processEventArray(obj)
-			if err != nil {
-				return err
-			}
-		case map[string]interface{}:
-			response = obj
-			if in.config.JSONObjects == "" {
-				mm, err = in.processEventArray([]interface{}{obj})
-				if err != nil {
-					return err
-				}
-			} else {
-				v, err = common.MapStr(obj).GetValue(in.config.JSONObjects)
-				if err != nil {
-					if err == common.ErrKeyNotFound {
-						break
-					}
-					return err
-				}
-				switch ts := v.(type) {
-				case []interface{}:
-					mm, err = in.processEventArray(ts)
-					if err != nil {
-						return err
-					}
-				default:
-					return errors.Errorf("content of %s is not a valid array", in.config.JSONObjects)
-				}
-			}
-		default:
-			in.log.Debug("http.response.body is not a valid JSON object", string(responseData))
-			return errors.Errorf("http.response.body is not a valid JSON object, but a %T", obj)
-		}
-
-		if mm != nil && in.config.Pagination.IsEnabled() {
-			if in.config.Pagination.Header != nil {
-				// Pagination control using HTTP Header
-				url, err := getNextLinkFromHeader(header, in.config.Pagination.Header.FieldName, in.config.Pagination.Header.RegexPattern)
-				if err != nil {
-					return errors.Wrapf(err, "failed to retrieve the next URL for pagination")
-				}
-				if ri.URL == url || url == "" {
-					in.log.Info("Pagination finished.")
-					break
-				}
-				ri.URL = url
-				if err = in.applyRateLimit(ctx, header, in.config.RateLimit); err != nil {
-					return err
-				}
-				in.log.Info("Continuing with pagination to URL: ", ri.URL)
-				continue
-			} else {
-				// Pagination control using HTTP Body fields
-				ri, err = createRequestInfoFromBody(in.config.Pagination, common.MapStr(response), common.MapStr(mm), ri)
-				if err != nil {
-					return err
-				}
-				if ri == nil {
-					break
-				}
-				if err = in.applyRateLimit(ctx, header, in.config.RateLimit); err != nil {
-					return err
-				}
-				in.log.Info("Continuing with pagination to URL: ", ri.URL)
-				continue
-			}
+		switch url.Scheme {
+		case "https":
+			return "443"
 		}
-		break
-	}
+		return "80"
+	}()
 
-	if mm != nil && in.config.DateCursor.IsEnabled() {
-		in.advanceCursor(common.MapStr(mm))
+	_, err = net.DialTimeout("tcp", fmt.Sprintf("%s:%s", url.Hostname(), port), time.Second)
```

Comment:
Should this use `net.JoinHostPort` to ensure IPv6 addresses are boxed? I think url.Hostname() strips brackets.